packages/common/src/models/Grant.ts

```diff
@@ -8,7 +8,7 @@ export type Grant = {
   grantee_address: string
   user_id: Nullable<ID>
   is_revoked: boolean
-  is_approved: boolean
+  is_approved: boolean | null
```

Comment:
Let's chat about this. The APIs for managed users don't make this distinction and the SDK types currently say this field is guaranteed to be a boolean.

Reply:
ooh gotcha. we should fix those because
`null` means request has not been rejected or accepted
`true` means explicitly accepted by manager
`false` means explicitly rejected by manager

It was designed this way because I originally thought we would need to distinguish between "pending" and "rejected" (now we don't because we decided to allow rejected requests to be re-sent immediately). But I think it's OK to leave it like this, to keep the door open in the future to disallow spammy accounts from re-sending manager requests that were explicitly rejected.

Reply:
Do devapp grants use this field?
If not, I would maybe advocate that instead of a tri-state boolean, we use an explicit `state` field that can clearly indicate the difference between 'pending', 'accepted', 'rejected', 'revoked'. But if that would also break devapp grants, then we can update the API to use `null ` as a special value.
Just don't love special derived meaning on fields.